test/config/config_settings_spec.py

```diff
@@ -56,11 +59,41 @@
                 "export_note_sort_methods": ["notemodel", "guid"]
             }
 
-            config = ConfigSettings(mw.addonManager, settings)
-
-            config.save()
+            config = ConfigSettings(init_values=settings)
 
             assert config._config == settings
 
             for key in settings:
                 assert settings[key] == getattr(config, key)
+
+        with it("tries to save new values to config dictionary"):
+            addon_manager_mock = MagicMock()
+
+            old_settings = {
+                "snapshot_path": "",
+                "automated_snapshot": False,
+                "snapshot_root_decks": [],
+                "export_notes_reverse_order": False,
+                "export_note_sort_methods": []
+            }
+
+            new_settings = {
+                "snapshot_path": "testing",
+                "automated_snapshot": True,
+                "snapshot_root_decks": ["TestDeck1", "Other"],
+                "export_notes_reverse_order": True,
+                "export_note_sort_methods": ["notemodel", "guid"]
+            }
+
+            config = ConfigSettings(addon_manager=addon_manager_mock, init_values=old_settings)
+
+            for key in new_settings:
+                setattr(config, key, new_settings[key])
+
+            config.save()
+
+            assert config._config == new_settings
+            addon_manager_mock.writeConfig.assert_called_once()
+
+            for key in new_settings:
+                assert new_settings[key] == config._config[key]
```

Comment:
this seems to duplicate the assertion above (of just comparing dicts)

Reply:
Looking online says you're right, that a straight up comparison `assert config._config == new_settings` is fine :+1: I am used to not trusting such comparisons, and doing extra work myself :sweat_smile: Deleted